transport/httptransport/auth_token_db.go

```diff
@@ -0,0 +1,62 @@
+package httptransport
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"golang.org/x/xerrors"
+)
+
+// ErrTokenNotFound is returned when an auth token is not found in the database
+var ErrTokenNotFound = fmt.Errorf("auth token not found")
+
+// AuthTokenDB keeps a database of auth tokens with associated data
+type AuthTokenDB struct {
+	ds datastore.Batching
+}
+
+func NewAuthTokenDB(ds datastore.Batching) *AuthTokenDB {
+	return &AuthTokenDB{
+		ds: namespace.Wrap(ds, datastore.NewKey("/auth-token")),
+	}
+}
+
+// Put adds some data to the DB and returns an auth token
+func (db *AuthTokenDB) Put(ctx context.Context, data []byte) (authtok string, e error) {
+	// Create a new auth token and add it to the datastore
+	authTokenBuff := make([]byte, 1024)
```

Comment:
Let's say 2048 bits to match a [standard RSA key](https://stackoverflow.com/questions/589834/what-rsa-key-length-should-i-use-for-my-ssl-certificates). So that's 2048 / 8 = 256 bytes